addons/stock/models/stock_inventory.py

```diff
@@ -200,13 +200,48 @@ def action_cancel_draft(self):
     @api.multi
     def action_start(self):
         for inventory in self:
-            vals = {'state': 'confirm', 'date': fields.Datetime.now()}
             if (inventory.filter != 'partial') and not inventory.line_ids:
-                vals.update({'line_ids': [(0, 0, line_values) for line_values in inventory._get_inventory_lines_values()]})
+                line_values_list = inventory._get_inventory_lines_values()
+                line_ids = []
+                for line_values in line_values_list:
+                    line_id = inventory._insert_inventory_line(line_values)
+                    line_ids.append(line_id)
+                lines = self.env['stock.inventory.line'].browse(line_ids)
+                lines.modified(self.env['stock.inventory.line']._fields)
+
+            vals = {'state': 'confirm', 'date': fields.Datetime.now()}
             inventory.write(vals)
+        self.recompute()
         return True
+
     prepare_inventory = action_start
 
+    def _insert_inventory_line(self, values):
+        row = self._inventory_line_values_to_row(values)
```

Comment:
if you have access to `write` on stock.inventory, but no access to `create` stock.inventoty.line, with this SQL methode you can do that.
maybe add `self.env['stock.inventory.line].check_access_right('create')`.

Reply:
@fmdl strictly speaking, you are right. But having `write` access on `stock.inventory` and no access to create `stock.inventory.line` simply does not make sense business-wise. As the method is not public, I don't see the point into enforcing such a security policy.